### PR TITLE
#17: Modal

### PR DESCRIPTION
**Pull request by gcandal**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-components-vue/issues/17 |
| Decisions |   |
| Animated GIF | https://recordit.co/egNI39w6Gl  |

### CODE REVIEW

File: vue/components/ui/modal-platforme/modal-platforme.vue [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+                        <img src="~./assets/close.svg" />
+                    </slot>
+                </div>
+                <h1 class="title" v-if="title">{{ title }}</h1>
+                <h2 class="sub-title" v-if="subTitle">{{ subTitle }}</h2>
+                <div class="modal-content">
+                    <slot />
+                </div>
+                <div class="buttons-container">
+                    <div
```
Comment (joamag) [inline vue/components/ui/modal-platforme/modal-platforme.vue]: this must be a button color component from the ripe-commons-vue

File: vue/components/ui/modal-platforme/modal-platforme.vue [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+                        class="button button-color button-color-secondary button-cancel"
+                        v-if="cancelText"
+                        v-on:click="cancel"
+                    >
+                        {{ cancelText }}
+                    </div>
+                    <div
```
Comment (joamag) [inline vue/components/ui/modal-platforme/modal-platforme.vue]: this must be a button color component

File: vue/components/ui/modal-platforme/modal-platforme.vue [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+    border-radius: 4px 4px 4px 4px;
+    box-shadow: 0px 0px 12px #2d2d2d;
+    display: inline-block;
+    max-height: 80%;
+    opacity: 1;
+    overflow-y: auto;
+    padding: 40px 40px 40px 40px;
+    position: relative;
+    top: 50%;
+    transform: translateY(-50%);
```
Comment (joamag) [inline vue/components/ui/modal-platforme/modal-platforme.vue]: No transform hack

File: vue/components/ui/modal-platforme/modal-platforme.vue [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+    box-sizing: border-box;
+    max-height: 90%;
+    max-width: 100%;
+    padding: 20px 10px 20px 10px;
+}
+
+.modal.fade-enter > .modal-container,
+.modal.fade-leave-to > .modal-container {
+    opacity: 0;
+    transform: translateY(-48%);
```
Comment (joamag) [inline vue/components/ui/modal-platforme/modal-platforme.vue]: no transform

File: vue/components/ui/modal-platforme/modal-platforme.vue [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+.modal.fade-enter > .modal-container > .modal-content,
+.modal.fade-leave-to > .modal-container > .modal-content {
+    opacity: 0;
+}
+
+.modal.fade-enter-to > .modal-container > .modal-content {
+    opacity: 1;
+}
+
+.modal > .modal-container > .buttons-container > .button {
```
Comment (joamag) [inline vue/components/ui/modal-platforme/modal-platforme.vue]: no css for buttons, a button should be an inline-block element by default

File: vue/components/ui/modal-platforme/modal-platforme.vue [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+    transition: opacity 0.25s ease-in-out 0.125s;
+}
+
+.modal.fade-enter > .modal-container > .modal-content,
+.modal.fade-leave-to > .modal-container > .modal-content {
+    opacity: 0;
+}
+
+.modal.fade-enter-to > .modal-container > .modal-content {
+    opacity: 1;
```
Comment (joamag) [inline vue/components/ui/modal-platforme/modal-platforme.vue]: animations should be more complex and defined by CSS

File: vue/components/ui/modal-platforme/modal-platforme.vue [hunk 7]
```diff
@@ -0,0 +1,289 @@
...
+        },
+        name: {
+            type: String,
+            mandatory: true
+        },
+        overlay: {
+            type: Boolean,
+            default: true
+        },
+        overlayLeave: {
```
Comment (joamag) [inline vue/components/ui/modal-platforme/modal-platforme.vue]: ??

File: vue/components/ui/modal-platforme/modal-platforme.vue [hunk 8]
```diff
@@ -0,0 +1,289 @@
...
+            this.$emit("enter_cancelled");
+        },
+        beforeLeave() {
+            this.$emit("before_leave");
+        },
+        leave() {
+            this.$emit("leave");
+        },
+        afterLeave() {
+            this.$emit("after_leave");
```
Comment (joamag) [inline vue/components/ui/modal-platforme/modal-platforme.vue]: no need for these events. Let's keep the code simple enough.

File: vue/components/ui/modal-platforme/modal-platforme.vue (hunk 7 above)
Comment (gcandal) [reply vue/components/ui/modal-platforme/modal-platforme.vue]: overlayLeave = true => clicking on the overlay closes the modal

**Pull request merged**